### DISCUSSION TITLE
nullable object

### DISCUSSION THREAD

**Discussion by mauriceoc** [mswjs/data#202] [Q&A]
Anyone know how to define a nullable object?

I have tried the following:

```javascript
{
      id: primaryKey(String),
      name: nullable(String), // this works
      someObject: nullable(Object) // this does not work, cannot update it
} 
```

Also this:

```javascript
{
      id: primaryKey(String),
      name: nullable(String), // this works
      someObject: nullable(() => ({
            foo: String,
            bar: String
      })) // this does not work, cannot update it
} 
```

**Comment by kettanaito** [comment]
Hey, @mauriceoc. There's no support for this as of now but we'd definitely have to support this. Looks like a matter of extending the `ModelValueType` properly and generating the nested object values for nested nullable objects (nested regular objects are already supported). I've opened a new issue about this in #203.

For now, you can move the nested object into a separate model and create a relationship:

```js
const db = factory({
  user: {
    id: primaryKey(String),
    permissions: nullable(oneOf('permission')),
  },
  permission: {
    id: primaryKey(String),
    title: String
  }
})
```
